lib/Ray/sources/Window.cpp

```diff
@@ -93,34 +94,48 @@ void RAY::Window::clear(const RAY::Color &color)
 	ClearBackground(color);
 }
 
-void RAY::Window::beginDrawing(void)
+void RAY::Window::setDrawingState(enum RAY::Window::drawingState state)
 {
-	::BeginDrawing();
-}
-
-void RAY::Window::endDrawing(void)
-{
-	::EndDrawing();
-}
-
-void RAY::Window::beginMode2D(Camera::Camera2D &camera)
-{
-	BeginMode2D(camera.getCamera());
+	if (state == this->_drawingState)
+		return;
+	switch (state)
+	{
+	case DRAWING:
+		BeginDrawing();
+		break;
+	case IDLE:
+		EndDrawing();
+		break;
+	}
+	this->_drawingState = state;
 }
 
-void RAY::Window::beginMode3D(Camera::Camera3D &camera)
+void RAY::Window::useCamera(RAY::Camera::Camera2D &camera)
 {
-	BeginMode3D(camera.getCamera());
+	this->_displayState = RAY::Window::TWO_DIMENSIONNAL;
```

Comment:
If a camera is already loaded I believe you should unuse the camera. If you don't you might have some leaks.